### PR TITLE
Disallow equal names in MetadataServiceSpec.

### LINKED ISSUE #2107: MetadataServiceSpec failing test
Saw this fail on a travis master build for `mongodb_3_0`. Commit 81bfcae.

```
[error] x and mounts when any children happen to be mount points (1 second, 95 ms)
[error]  Falsified after 6 passed tests.
[error]  > ARG_0: FileName(/v)
[error]  > ARG_1: DirName(6/j)
[error]  > ARG_2: DirName(p)
[error]  > ARG_3: FileName(.ni])
[error]  > ARG_4: DirName(p)
[error]  > ARG_5: (Fix(Select(SelectAll,List(Proj(Fix(Splice(None)),None)),Some(TableRelationAST(FileIn(DirIn(Current,DirName(mj)),FileName(b')),Some(Klbe))),Some(Fix(Binop(Fix(Ident(name, address)),Fix(Binop(Fix(Match(Fix(Ident(q: "a")),List(Case(Fix(SetLiteral(List(Fix(BoolLiteral(true)), Fix(FloatLiteral(9.622731996933524))))),Fix(Ident(h)))),None)),Fix(Switch(List(Case(Fix(InvokeFunction(CIName(interval),List(Fix(StringLiteral(PT1473H20M41.131S))))),Fix(InvokeFunction(CIName(timestamp),List(Fix(StringLiteral(2017-04-26T20:07:51.712Z)))))), Case(Fix(Ident(q: "a")),Fix(Ident(q: "a")))),None)),IndexDeref)),IndexDeref))),Some(GroupBy(List(Fix(Select(SelectDistinct,List(Proj(Fix(Splice(None)),None)),None,None,Some(GroupBy(List(Fix(Ident(name, address)), Fix(Ident(q: "a")), Fix(Vari(ixt))),Some(Fix(SetLiteral(List(Fix(NullLiteral()), Fix(StringLiteral(zGthnsyz)))))))),None))),Some(Fix(Binop(Fix(InvokeFunction(CIName(count),List(Fix(Binop(Fix(Ident(q: "a")),Fix(Vari(pvrc)),Ge))))),Fix(Vari(cetm)),Gt))))),Some(OrderBy(NonEmpty[(ASC,Fix(Switch(List(Case(Fix(Ident(q: "a")),Fix(InvokeFunction(CIName(timestamp),List(Fix(StringLiteral(2017-04-26T20:07:51.716Z)))))), Case(Fix(Ident(name, address)),Fix(Binop(Fix(InvokeFunction(CIName(length),List(Fix(Ident(j))))),Fix(Select(SelectDistinct,List(Proj(Fix(Splice(None)),None)),None,None,Some(GroupBy(List(Fix(InvokeFunction(CIName(interval),List(Fix(StringLiteral(PT2604H25M55.118S))))), Fix(InvokeFunction(CIName(interval),List(Fix(StringLiteral(PT2699H11M3.74S)))))),Some(Fix(Ident(name, address))))),Some(OrderBy(NonEmpty[(ASC,Fix(Vari(lsjd))),(ASC,Fix(Ident(q: "a"))),(ASC,Fix(InvokeFunction(CIName(oid),List(Fix(StringLiteral(123456)))))),(ASC,Fix(Vari(z)))])))),FieldDeref)))),Some(Fix(Binop(Fix(Ident(name, address)),Fix(Vari(u)),IndexDeref)))))),(DESC,Fix(Binop(Fix(Unop(Fix(Switch(List(Case(Fix(InvokeFunction(CIName(timestamp),List(Fix(StringLiteral(2017-04-26T20:07:51.721Z))))),Fix(Ident(q))), Case(Fix(Ident(g)),Fix(Ident(name, address))), Case(Fix(Ident(name, address)),Fix(Ident(name, address)))),None)),FlattenArrayValues)),Fix(InvokeFunction(CIName(like),List(Fix(Ident(name, address)), Fix(StringLiteral(B%)), Fix(StringLiteral())))),IfUndefined)))])))),Variables(Map(:ﾏ汗ᕈ኎ -> VarValue(null))))
[error]  > ARG_6: (FileSystemType(sM),ConnectionUri(hcck:cdzkppZ))
[error]  > '{"children":[{"name":".ni]","type":"file","mount":"view"},{"name":"/v","type":"file"},{"name":"6/j","type":"directory"},{"name":"p","type":"directory","mount":"module"}]}'
[error]  
[error]   is not equal to 
[error]  
[error]  '{"children":[{"mount":"view","type":"file","name":".ni]"},{"type":"file","name":"/v"},{"type":"directory","name":"6/j"},{"mount":"module","type":"directory","name":"p"},{"mount":"sM","type":"directory","name":"p"}]}' (MetadataServiceSpec.scala:139)
[error] Actual:   ...ren":{"[name]":"[.ni][mount]":"[view]"},{"[name":"/v","]ty...e"[}],[{]"n...:"[6]/[j]"[],[]"t...y"[}],[{]"n...:"[p]"[],[]"[]t[]yp...,"[]m[]ou...:"[mo]d[ul]e["}]}
[error] Expected: ...ren":{"[mount]":"[view]",...,"[name]":"[.ni][]ty...e"[],[]"n...:"[]/[v]"[}],[{]"t...y"[],[]"n...:"[6/j]"[}],[{]"[moun]t[":"module","t]yp...,"[na]m[e":"p"},{"m]ou...:"[sM","type":"]d[ir]e[ctory]
```

### PR DESCRIPTION
**Pull request by alissapajer**
Fixes #2107.

### CODE REVIEW

File: web/src/test/scala/quasar/api/services/MetadataServiceSpec.scala [hunk 1]
```diff
@@ -121,19 +121,19 @@ class MetadataServiceSpec extends quasar.Qspec with FileSystemFixture with Http4
         moduleName: DirName,
         vcfg: (Fix[Sql], Variables),
         fsCfg: (FileSystemType, ConnectionUri)
-      ) => (fileName ≠ viewName && directoryName ≠ fsMountName) ==> {
+      ) => (fileName ≠ viewName && directoryName ≠ fsMountName && fsMountName ≠ moduleName) ==> {
```
Comment (djspiewak) [inline web/src/test/scala/quasar/api/services/MetadataServiceSpec.scala]: Normally, I would prefer newtyping and constructing a generator which simply doesn't produce the errant values.  But in this case, the bad values are so sparse in the domain that we probably don't need to worry about it.

File: web/src/test/scala/quasar/api/services/MetadataServiceSpec.scala [hunk 2]
```diff
@@ -121,19 +121,19 @@ class MetadataServiceSpec extends quasar.Qspec with FileSystemFixture with Http4
...
         val moduleConfig: List[Statement[Fix[Sql]]] = List(
           FunctionDecl(CIName("FOO"), List(CIName("Bar")), Fix(boolLiteral(true))))
         val parent: ADir = rootDir </> dir("foo")
         val mnts = Map[APath, MountConfig](
-          (parent </> file(viewName.value),   MountConfig.viewConfig(vcfg)),
-          (parent </> dir(fsMountName.value), MountConfig.fileSystemConfig(fsCfg)),
-          (parent </> dir1(moduleName),       MountConfig.moduleConfig(moduleConfig)))
+          (parent </> file1(viewName), MountConfig.viewConfig(vcfg)),
+          (parent </> dir1(fsMountName), MountConfig.fileSystemConfig(fsCfg)),
+          (parent </> dir1(moduleName), MountConfig.moduleConfig(moduleConfig)))
```
Comment (djspiewak) [inline web/src/test/scala/quasar/api/services/MetadataServiceSpec.scala]: I very much approve of the unaligning.  :-P

File: web/src/test/scala/quasar/api/services/MetadataServiceSpec.scala [hunk 3]
```diff
@@ -121,19 +121,19 @@ class MetadataServiceSpec extends quasar.Qspec with FileSystemFixture with Http4
...
         val mem = InMemState fromFiles Map(
-          (parent </> file(fileName.value), Vector()),
-          (parent </> dir(directoryName.value) </> file("quux"), Vector()),
-          (parent </> file(viewName.value), Vector()),
-          (parent </> dir(fsMountName.value) </> file("bar"), Vector()))
+          (parent </> file1(fileName), Vector()),
+          (parent </> dir1(directoryName) </> file("quux"), Vector()),
+          (parent </> file1(viewName), Vector()),
+          (parent </> dir1(fsMountName) </> file("bar"), Vector()))
```
Comment (djspiewak) [inline web/src/test/scala/quasar/api/services/MetadataServiceSpec.scala]: Were these tests just using the wrong names before?  I don't know anything about this code, it just seems weird that the `1` suffix is used now and wasn't used before, but no new declarations were added.

Comment (alissapajer) [inline web/src/test/scala/quasar/api/services/MetadataServiceSpec.scala]: [`dir`](https://github.com/slamdata/scala-pathy/blob/946c304b09777b6f67257b6b1a648bd34e568a18/core/src/main/scala/pathy/Path.scala#L110) calls directly into `dir1` so by using `dir` we were unwrapping and then rewrapping `DirName`. So this had nothing to do with the test fix; just avoiding that extra step.

Comment (wemrysi) [inline web/src/test/scala/quasar/api/services/MetadataServiceSpec.scala]: @djspiewak note the lack of `.value` from all the arguments, which was unwrapping the underlying `String` whereas the `1`-suffixed versions operate on `FileName`s and `DirName`s directly.

File: web/src/test/scala/quasar/api/services/MetadataServiceSpec.scala (hunk 1 above)
Comment (alissapajer) [inline web/src/test/scala/quasar/api/services/MetadataServiceSpec.scala]: Yeah, we'll have to generate very few extra values. This test was only sporadically failing in travis.

File: web/src/test/scala/quasar/api/services/MetadataServiceSpec.scala (hunk 3 above)
Comment (alissapajer) [inline web/src/test/scala/quasar/api/services/MetadataServiceSpec.scala]: @wemrysi I think we answered at exactly the same time! :)

**Comment by djspiewak** [comment]
@alissapajer The test failure here is in `quasar.ViewReadQueryRegressionSpec`, which seems unrelated?

**Comment by alissapajer** [comment]
@djspiewak yeah, that error is unrelated. But I've been waiting for that sort of CB error to pop up again so I could open a ticket. So I'll restart that job after I get the ticket open.

**Comment by alissapajer** [comment]
@djspiewak this should be good to go!

**Pull request merged**